src/Hosting/PageMatcherPolicy.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2014-2020 Sarin Na Wangkanai, All Rights Reserved.
+// The Apache v2. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Matching;
+
+namespace Wangkanai.Detection.Hosting
+{
+    internal class ResponsivePageMatcherPolicy : MatcherPolicy, IEndpointComparerPolicy, IEndpointSelectorPolicy
+    {
+        public ResponsivePageMatcherPolicy() 
+            => Comparer = EndpointMetadataComparer<IResponsiveMetadata>.Default;
```

Comment:
This is another thing that i need to investigate more `EndpointMetadataComparer` 

https://github.com/dotnet/aspnetcore/blob/master/src/Http/Routing/src/Matching/EndpointMetadataComparer.cs